src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/DGSpecFactory.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat.Commands.Package.Update
+{
+    internal class DGSpecFactory : IDGSpecFactory
+    {
+        IEnvironmentVariableReader _environmentVariableReader;
+
+        public DGSpecFactory()
+        {
+            _environmentVariableReader = new EnvironmentVariableWrapper();
+        }
+
+        public DependencyGraphSpec GetDependencyGraphSpec(string project)
+        {
+            string tempFile = Path.GetTempFileName();
+            try
+            {
+                if (!RunMsbuildTarget(project, tempFile))
+                {
+                    return null;
+                }
+
+                DependencyGraphSpec result = DependencyGraphSpec.Load(tempFile);
+
+                return result;
+            }
+            finally
+            {
+                File.Delete(tempFile);
+            }
+        }
+
+        private bool RunMsbuildTarget(string project, string tempFile)
+        {
+            // When being run from the dotnet CLI, use the same dotnet executable, just in case the dotnet on the PATH is different
+            // But when NuGet.CommandLine.XPlat is being called directly, call dotnet on the path, so this code is debuggable.
+            string dotnetPath = _environmentVariableReader.GetEnvironmentVariable("DOTNET_HOST_PATH") ?? "dotnet";
+
+            // don't redirect stdout or stderr, so errors are output. But use quiet verbosity, so that success has no output.
+            ProcessStartInfo processStartInfo = new ProcessStartInfo(dotnetPath)
+            {
+                Arguments = $"msbuild " +
+                $"\"{project}\" " +
+                $"-restore:false " +
+                $"-target:GenerateRestoreGraphFile " +
+                $"-property:RestoreGraphOutputPath=\"{tempFile}\" " +
+                $"-property:RestoreRecursive=false " +
+                $"-nologo " +
+                $"-verbosity:quiet " +
+                $"-tl:false " +
```

Comment:
Question - Do we always need to disable the terminal logger or is that just because of the bug you recently told me about?

Reply:
The terminal logger always logs errors, so it's probably fine to allow the terminal logger. But it always writes a single empty line, which annoyed me, so I disabled it so that the msbuild child process won't write any output, not even an empty line, when it works successfully.

Reply:
You're right it'll auto turn off when there's no interactive console, like in our tests. But if you run update package in an interactive terminal, since we don't redirect stdout for the msbuild child process, it'll have direct output access to the terminal and detect it does support colours (and all other VT100 control codes).

This is good, because if generating the DGSpec fails, the error will be red.